### DISCUSSION TITLE
Support upload files

### DISCUSSION THREAD

**Discussion by yuri2peter** [mdx-editor/editor#233] [Ideas]
Allow users to drag and drop files (not just images) and insert a download link.
Perhaps we should consider combining the file and image upload handlers (considering we already have an image uploader).
A convenient way to achieve this could be to let developers handle what the editor should insert after each file(image or file) is uploaded.

**Comment by noteverseio** [comment]
This can be done via a plugin, you can implement the plugin using the MDXEditor extensions, e.g. directives, code block or jsx component, you can customize you file upload logic in your extension.

One example is you can implement a FileUploader JSX component which handles the file upload logic.

**Reply by gradywetherbee** [reply]
At some point it would be great to see an example of this in the docs when possible - for noobs like me
